presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -333,6 +326,11 @@ private ShardInfo createShardInfo(UUID shardUuid, File file, Set<String> nodes,
             return shardDelta(shardUuid, Optional.empty());
         }
 
+        if (isBackupAvailable()) {
```

Comment:
This is what led me to think if we should have a `NullBackupStore` that way we always submit the backup and if it is a ``NullBackupStore`` it does nothing. If we decide to do this, it should be the behavior everywhere. 
It is weird if all operations except this one check `isBackupAvailable`